lisa-utils/src/main/scala/lisa/utils/TheoriesHelpers.scala

```diff
@@ -61,6 +61,47 @@ trait TheoriesHelpers extends KernelHelpers {
      * a Theorem with a given name or a Definition with a the given name as symbol
      */
     def getJustification(name: String): Option[theory.Justification] = theory.getAxiom(name).orElse(theory.getTheorem(name)).orElse(theory.getDefinition(name))
+
+    /**
+     * Verify if a given formula belongs to some language
+     *
+     * @param phi The formula to check
+     * @return The List of undefined symols
+     */
+    def findUndefinedSymbols(phi: Formula): Set[ConstantLabel] = phi match {
+      case PredicateFormula(label, args) =>
+        label match {
+          case l: ConstantPredicateLabel => ((if (theory.isSymbol(l)) Nil else List(l)) ++ args.flatMap(findUndefinedSymbols)).toSet
```

Comment:
If the top predicate label is a known symbol, does that guarantee all its children are? Is there another check on this somewhere?

Reply:
my bad, misread